include/gsl/util

```diff
@@ -52,40 +52,27 @@ template <class F>
 class final_action
 {
 public:
-    static_assert(!std::is_reference<F>::value && !std::is_const<F>::value &&
-                      !std::is_volatile<F>::value,
-                  "Final_action should store its callable by value");
+    explicit final_action(F f_) : f(std::move(f_)) { }
+    ~final_action() { if (invoke) f(); }
 
-    explicit final_action(F f) noexcept : f_(std::move(f)) {}
+    final_action(final_action&& other)
+        : f(std::move(other.f)), invoke(std::exchange(other.invoke, false))
+    { }
 
-    final_action(final_action&& other) noexcept
-        : f_(std::move(other.f_)), invoke_(std::exchange(other.invoke_, false))
-    {}
-
-    final_action(const final_action&) = delete;
+    final_action(const final_action& rhs) = delete;
     final_action& operator=(const final_action&) = delete;
-    final_action& operator=(final_action&&) = delete;
-
-    // clang-format off
-    GSL_SUPPRESS(f.6) // NO-FORMAT: attribute // terminate if throws
-    // clang-format on
-    ~final_action() noexcept
-    {
-        if (invoke_) f_();
-    }
+    final_action& operator=(final_action&& other) = delete;
 
 private:
-    F f_;
-    bool invoke_{true};
+    F f;
+    bool invoke = true;
 };
 
 // finally() - convenience function to generate a final_action
 template <class F>
-GSL_NODISCARD final_action<typename std::remove_cv<typename std::remove_reference<F>::type>::type>
-finally(F&& f) noexcept
+auto finally(F f) noexcept
 {
-    return final_action<typename std::remove_cv<typename std::remove_reference<F>::type>::type>(
-        std::forward<F>(f));
+    return final_action<F>{f};
```

Comment:
All `final_action` members are `noexcept(false)`, so having `noexcept` here looks wrong. Yet in my opinion it would be better to have `final_action` `noexcept` depending on `F`.